### DISCUSSION TITLE
how to use old style backup data

### DISCUSSION THREAD

**Discussion by UKONNsoft** [massCodeIO/massCode#68] [Q&A] [answered]
I have backup data looks like the below.
<img width="163" alt="スクリーンショット 2022-05-04 15 44 46" src="https://user-images.githubusercontent.com/7405040/166633835-acd7c5e4-ebe1-4e7a-ab45-a66056718780.png">
But current massCode(v2.5.0) doesn't seem to load this style of data.
I want massCode to load it.
PLS tell me how to do...

**Comment by antonreshetov** [comment, marked as answer]
@UKONNsoft massCode 2.0 has no mechanism for creating a backup. Also after migration from 1.0 valid database file is `db.json`.

For now, you can create a manual backup by creating a copy of the current database somewhere.

**Reply by UKONNsoft** [reply]
thank you,
You mean I have to convert my backup data which had been used for former version to db.json manually?
I'm afraid it's too hard.
Is former version of massCode avairable? I mean downgrade massCode to utilize my old data.

**Reply by antonreshetov** [reply]
The point of a backup is to roll back if something went wrong, if you are using 2.0 then you migrated from 1.0 so you have the latest version of the data. You can also choose a backup folder when you migrate, thereby migrating to 2.0 the selected version of the database.

**Reply by bompi** [reply]
Hello.
I have a db.json file from my last backup with Masscode 2.0.0 b1
I just installed Masscode 2.6.1 but it seems unable to read my db.json file.
Is there any modification to run on the data to make it readable again ?
Thanks in advance

**Comment by bompi** [comment]
I manually modified the old JSON file to make it compliant with the new version.
Now hopefully it seems to be OK. ;-)
